macros/edr/materializations/tests/test.sql

```diff
@@ -0,0 +1,243 @@
+{% macro query_test_result_rows(sample_limit=none) %}
+  {%- set query -%}
+      select * from ({{ sql }}) {%- if sample_limit -%} limit {{ sample_limit }} {%- endif -%}
+  {%- endset -%}
+  {% do return(elementary.agate_to_dicts(dbt.run_query(query))) %}
+{% endmacro %}
+
+{% macro cache_elementary_test_results_rows(elementary_test_results_rows) %}
+  {% do elementary.get_cache("elementary_test_results").update({model.unique_id: elementary_test_results_rows}) %}
+{% endmacro %}
+
+
+{% macro handle_anomaly_test(flattened_test) %}
+  {% set metrics_tables_cache = elementary.get_cache("tables").get("metrics") %}
+  {% set metrics_table = elementary.get_elementary_test_table(flattened_test.name, 'metrics') %}
+  {% if metrics_table %}
+    {% do metrics_tables_cache.append(metrics_table) %}
+  {% endif %}
+
+  {% set anomaly_scores_rows = elementary.query_test_result_rows() %}
+  {% if anomaly_scores_rows and flattened_test.short_name == "dimension_anomalies" %}
+    {% set sample_row = anomaly_scores_rows[0] %}
+    {% do elementary.cache_elementary_test_results_rows([elementary.get_anomaly_test_result_row(flattened_test, sample_row, anomaly_scores_rows)]) %}
+  {% else %}
+    {% set test_result_metrics_rows = {} %}
+    {% for anomaly_score_row in anomaly_scores_rows %}
+      {% do test_result_metrics_rows.setdefault(anomaly_score_row.metric_name, []) %}
+      {% do test_result_metrics_rows[anomaly_score_row.metric_name].append(anomaly_score_row) %}
+    {% endfor %}
+
+    {% set elementary_test_results_rows = [] %}
+    {% for metric_rows in test_result_metrics_rows.values() %}
+      {% for metric_row in metric_rows %}
+        {% do elementary_test_results_rows.append(elementary.get_anomaly_test_result_row(flattened_test, metric_row, metric_rows)) %}
```

Comment:
Should there actually be more than one row per metric?
(a bit strange to me that we add multiple rows for the same metric with the same full sample, but I'm guessing that doesn't really happen)

Reply:
The `all_columns_anomalies` test includes a row per column per metric, in which case it's necessary to do so.

Reply:
👍